static/js/vue-cdr-access/src/components/browseDisplay.vue

```diff
@@ -0,0 +1,255 @@
+                </browse-sort>
+            </div>
+        </div>
+        <div class="columns">
+            <div class="column is-10 spacing">
+                <p :class="{ no_results: numberOfRecords === 0}">
+                    There are <strong>{{ numberOfRecords }}</strong> {{ childTypeText }} in this level.
+                </p>
+                <p v-if="numberOfRecords > 0">
+                    Displaying <strong>{{ pagination_settings.start + 1}}</strong> to
+                    <strong>{{ displayNumber }}</strong> of <strong>{{ numberOfRecords }}</strong>
+                </p>
+            </div>
+            <div class="column is-2">
+                <modal-metadata  v-if="numberOfRecords > 0" :metadata="container_metadata"></modal-metadata>
+            </div>
+        </div>
+        <div class="columns">
+            <div class="column is-12">
+                <ul v-if="numberOfRecords > 0">
+                    <li class="column" :class="column_size" v-for="record in displayList"
+                        :key="record.id">
+                        <a :href="record.uri">
+                            <i class="fa" :class="recordType(record.type)"></i>
+                            <div class="record-count">{{ recordCountFormat(record.counts.child) }}</div>
+                            <div class="record-title">{{ record.title }}</div>
+                        </a>
+                    </li>
+                </ul>
+            </div>
+        </div>
+        <pagination :per-page="recordsPerPage"
+                    :number-of-records="numberOfRecords"
+                    :page-base-url="container_metadata.uri"
+                    @pagination-records-to-display="pageToDisplay">
+        </pagination>
+    </div>
+</template>
+
+<script>
+    import browseSearch from './browseSearch.vue';
+    import browseSort from './browseSort.vue';
+    import modalMetadata from './modalMetadata.vue';
+    import pagination from './pagination.vue';
+    import debounce from 'lodash.debounce';
+
+    export default {
+        name: 'browseDisplay',
+
+        components: {
+            browseSearch,
+            browseSort,
+            modalMetadata,
+            pagination
+        },
+
+        props: {
+            browsePath: String,
+            recordsPerPage: {
+                type: Number,
+                default: 20
+            }
+        },
+
+        data() {
+            return {
+                column_size: 'is-3',
+                container_metadata: {},
+                pagination_settings: {},
+                record_list: []
+            }
+        },
+
+        computed: {
+            childTypeText() {
+                if (this.container_metadata.type === 'AdminUnit') {
+                    return 'collections';
+                } else {
+                    return 'items';
+                }
+            },
+
+            displayList() {
+                return this.record_list.slice(
```

Comment:
I know i asked about this on the previous PR, but is this component asking the server for each page, or is it trying to slice up a single list that is retrieved once?

Reply:
The list is only retrieved once. Then creates a new array from the slice with just the needed records. We could drop this if we pulled new data from the server on each page change

Reply:
I think we should do that, we are going to need to pull in one page of results at a time, both with sorting and pagination. I'm realizing that even with admin sets, we have over the top cases like UARMS with its 92 collections, which is more than the page size allowed by the server at the moment.